Added Languages section to skills and changed formatting of lists to …

Tcintra opened the pull request.

Tcintra committed:
Added Languages section to skills and changed formatting of lists to a flexbox

Tcintra committed:
Steps 1 and 2 of the comments walkthrough. Add temporary new tab on navbar that directs to the /data URL and a button on the front page that uses fetch() to generate a quote from the /data servlet.

Tcintra committed:
Step 3 of the walkthrough: Add Gson dependency to pom.xml makefile, modify DataServlet to handle the JSON hardcoded comment. Modify index.html to contain a separate button that calls the getJSONString() method in script.js to fetch the hardcoded comments from the Servlet and display them in the DOM as an unordered list.

Tcintra committed:
Step 4 of Walkthrough. Add java file to generate a comment section class with history and totalComments attributes. Modify servlet to generate a comment section that stores all previous comments and displays them on the DOM, as well as a count of how many comments have been made. Modify the javascript to contain a getCommentSection() function that fetches data from the servlet to display on the DOM. Add the <form> with the history, totalComments, and an input field.

Tcintra commented:
This pull request contains steps 1,2,3,4  of the week-3 walkthrough, as well as a tiny modification I did to the website over the weekend.

wulian1223 commented on portfolio/src/main/webapp/script.js:
```diff
@@ -11,3 +11,60 @@
+ * Fetches stats from the servers and adds them to the DOM.
+ */
+function getJSONString() {
+  fetch('/data').then(response => response.json()).then((myString) => {
```
if `myString` is object, update the name. e.g  `comments`

Tcintra replied:
Updated the myString object to a Comment object!

wulian1223 commented on portfolio/src/main/webapp/script.js:
```diff
@@ -11,3 +11,60 @@
+    const stringsListElement = document.getElementById('JSON-strings-container');
+    stringsListElement.innerHTML = '';
+    
+    stringsListElement.appendChild(
```
use for loop instead of hardcoded

Tcintra replied:
I no longer have any hard coded comments, I am now loading stuff from the datastorage

wulian1223 commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -13,20 +13,57 @@
+    // Get the input from the form.
+    String commentString = request.getParameter("comment");
+
+    return commentString;
```
Optional, you can return `request.getParameter("comment");` directly

Tcintra replied:
done! I am using getParameter() directly

Tcintra committed:
Add dependency to pom.xml to handle datastorage. Rename CommentSection.java class to Comment.java to handle comment objects. Modify DataServlet.java to generate new comments and store them in the datastore. Add LoadCommentsServlet.java to load the comment history using queries. Comments are stored as an ArrayList<Comment> and are added to the DOM onload.

Tcintra commented:
I deleted the stuff from steps 1-3 so I no longer have a myString object. Instead, I created a comment class that holds the data for each comment: "id", "body", "timeStamp", and I load that instead. I also deleted all the hard-coded stuff, instead I now have an input field for comments which get displayed as an unordered list from the comments JSON file. I just pushed these changes, which address steps 4 and 5 in the walkthrough. Right now, there should be a very rudimentary comment section with the history of all the comments made, stored in DataStore.

serzhk commented:
Thomas, is this comment resolving Lian's comments? Could you reply to each
separate comment with done please...

On Mon, Jun 1, 2020 at 2:39 PM Tcintra <notifications@github.com> wrote:

> I deleted the stuff from steps 1-3 so I no longer have a myString object.
> Instead, I created a comment class that holds the data for each comment:
> "id", "body", "timeStamp", and I load that instead. I also deleted all the
> hard-coded stuff, instead I now have an input field for comments which get
> displayed as an unordered list from the comments JSON file. I just pushed
> these changes, which address steps 4 and 5 in the walkthrough. Right now,
> there should be a very rudimentary comment section with the history of all
> the comments made, stored in DataStore.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/Tcintra/step/pull/6#issuecomment-637125743>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APUURBCMHHSADYPITQ65FATRUQNXXANCNFSM4NOMQYSQ>
> .
>
